### PR TITLE
hardening/1313_flip_coordinates_swap_coordinates

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #1313 
* 100% tests passed (`cygnus-ngsi`):
```
Tests run: 266, Failures: 0, Errors: 0, Skipped: 0
```

Relevant tests for this PR:
```
[NGSICartoDBSink.configure] ----------------------------------------- Configured 'swap_coordinates' cannot be different than 'true' or 'false'
[NGSICartoDBSink.configure] ----------------------------------  OK  - 'swap_coordinates=falso' was detected
[CartoDBAggregator.aggregate] --------------------------------------- When aggregating a single geolocated event, if swap_coordinates=true then the_geom field contains a point with exchanged latitude and longitude.
[CartoDBAggregator.aggregate] --------------------------------  OK  - '('2016-04-20T07:19:55.801Z','somePath','someId','someType',ST_SetSRID(ST_MakePoint(40.3833,-3.7167), 4326),'someValue2','[]')' contains the coordinates '-3.7167, 40.3833' swapped
```

* (Unofficial) e2e tests passed:

Swapping:

```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "speed",            "type" : "float",            "value" : "78"          },          {            "name" : "the_geom",            "type" : "geometry",            "value" : "40.361, -3.4099",            "metadatas": [              {                "name": "location",                "type": "string",                "value": "WGS84"              }            ]          }        ],        "type" : "car",        "isPattern" : "false",        "id" : "car1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
...
Persisting data at NGSICartoDBSink. Schema (xxx), Table (x002ftrafficxffffcar1xffffcar), Data (('2016-11-28T15:17:54.561Z','/traffic','car1','car',ST_SetSRID(ST_MakePoint(-3.4099,40.361), 4326),'78','[]'))
```
Not swapping:

```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "speed",            "type" : "float",            "value" : "78"          },          {            "name" : "the_geom",            "type" : "geometry",            "value" : "40.361, -3.4099",            "metadatas": [              {                "name": "location",                "type": "string",                "value": "WGS84"              }            ]          }        ],        "type" : "car",        "isPattern" : "false",        "id" : "car1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
...
Persisting data at NGSICartoDBSink. Schema (xxx), Table (x002ftrafficxffffcar1xffffcar), Data (('2016-11-28T15:19:01.131Z','/traffic','car1','car',ST_SetSRID(ST_MakePoint(40.361,-3.4099), 4326),'78','[]'))
```
"Flipping":

```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "speed",            "type" : "float",            "value" : "78"          },          {            "name" : "the_geom",            "type" : "geometry",            "value" : "40.361, -3.4099",            "metadatas": [              {                "name": "location",                "type": "string",                "value": "WGS84"              }            ]          }        ],        "type" : "car",        "isPattern" : "false",        "id" : "car1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
...
Persisting data at NGSICartoDBSink. Schema (soyunpaquete), Table (x002ftrafficxffffcar1xffffcar), Data (('2016-11-29T08:48:39.928Z','/traffic','car1','car',ST_SetSRID(ST_MakePoint(-3.4099,40.361), 4326),'78','[]'))
```

Not "flipping":

```
Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "speed",            "type" : "float",            "value" : "78"          },          {            "name" : "the_geom",            "type" : "geometry",            "value" : "40.361, -3.4099",            "metadatas": [              {                "name": "location",                "type": "string",                "value": "WGS84"              }            ]          }        ],        "type" : "car",        "isPattern" : "false",        "id" : "car1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
...
Persisting data at NGSICartoDBSink. Schema (soyunpaquete), Table (x002ftrafficxffffcar1xffffcar), Data (('2016-11-29T08:49:10.99Z','/traffic','car1','car',ST_SetSRID(ST_MakePoint(40.361,-3.4099), 4326),'78','[]'))
```

### CODE REVIEW

File: cygnus-ngsi/conf/agent_ngsi.conf.template [hunk 1]
```diff
@@ -386,7 +386,7 @@ cygnus-ngsi.sinks.cartodb-sink.channel = cartodb-channel
 # absolute path to the CartoDB file containing the mapping between FIWARE service/CartoDB usernames and CartoDB API Keys
 cygnus-ngsi.sinks.cartodb-sink.keys_conf_file = /usr/cygnus/conf/cartodb_keys.conf
 # if true the latitude and longitude values are exchanged, false otherwise
-#cygnus-ngsi.sinks.cartodb-sink.flip_coordinates = true
+#cygnus-ngsi.sinks.cartodb-sink.swap_coordinates = true
```
Comment (fgalan) [inline cygnus-ngsi/conf/agent_ngsi.conf.template]: Potential backward compatibility problem here? Or this feature has been never released?

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md [hunk 2]
```diff
@@ -444,7 +444,7 @@ cygnus-ngsi.sinks.cartodb-sink.enable_grouping = false
 cygnus-ngsi.sinks.cartodb-sink.enable_name_mappings = false
 cygnus-ngsi.sinks.cartodb-sink.enable_lowercase = false
 cygnus-ngsi.sinks.cartodb-sink.keys_conf_file = /usr/cygnus/conf/cartodb_keys.conf
-cygnus-ngsi.sinks.cartodb-sink.flip_coordinates = true
+cygnus-ngsi.sinks.cartodb-sink.swap_coordinates = true
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Same

File: doc/cygnus-ngsi/installation_and_administration_guide/ngsi_agent_conf.md [hunk 3]
```diff
@@ -376,7 +376,7 @@ cygnus-ngsi.sinks.cartodb-sink.channel = cartodb-channel
 # absolute path to the CartoDB file containing the mapping between FIWARE service/CartoDB usernames and CartoDB API Keys
 cygnus-ngsi.sinks.cartodb-sink.keys_conf_file = /usr/cygnus/conf/cartodb_keys.conf
 # if true the latitude and longitude values are exchanged, false otherwise
-#cygnus-ngsi.sinks.cartodb-sink.flip_coordinates = true
+#cygnus-ngsi.sinks.cartodb-sink.swap_coordinates = true
```
Comment (fgalan) [inline doc/cygnus-ngsi/installation_and_administration_guide/ngsi_agent_conf.md]: Same

Comment (frbattid) [inline doc/cygnus-ngsi/installation_and_administration_guide/ngsi_agent_conf.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1329/commits/423f020dace40fa1f12f7ec25758fe3fcc3aee13

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md (hunk 2 above)
Comment (frbattid) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1329/commits/423f020dace40fa1f12f7ec25758fe3fcc3aee13

File: cygnus-ngsi/conf/agent_ngsi.conf.template (hunk 1 above)
Comment (frbattid) [inline cygnus-ngsi/conf/agent_ngsi.conf.template]: Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1329/commits/423f020dace40fa1f12f7ec25758fe3fcc3aee13

**Comment by fgalan** [comment]
LGTM

**Pull request merged**